streams/service/src/main/java/com/hortonworks/streamline/streams/service/UDFCatalogResource.java

```diff
@@ -432,10 +432,11 @@ private void validateUDF(Set<String> udfs, UDF udf, boolean checkDuplicate) {
     }
 
     private void checkDuplicate(UDF udf) {
-        Collection<UDF> existing = catalogService.listUDFs(
-                Collections.singletonList(new QueryParam(UDF.NAME, udf.getName())));
+        List<QueryParam> qps = QueryParam.params(UDF.NAME, udf.getName(), UDF.CLASSNAME, udf.getClassName(),
+                UDF.TYPE, udf.getType().toString());
+        Collection<UDF> existing = catalogService.listUDFs(qps);
         if (!existing.isEmpty()) {
-            LOG.warn("UDF with name {} already exists", udf.getName());
+            LOG.warn("UDF with same (name, classname, type) already exists", udf);
```

Comment:
Maybe missing `{}`? Regardless of that let's add space between  `exists` and udf.